DataFormats/PortableTestObjects/interface/alpaka/TestDeviceCollection.h

```diff
@@ -0,0 +1,22 @@
+#ifndef DataFormats_PortableTestObjects_interface_alpaka_TestDeviceCollection_h
+#define DataFormats_PortableTestObjects_interface_alpaka_TestDeviceCollection_h
+
+#include "DataFormats/Portable/interface/alpaka/PortableCollection.h"
+#include "DataFormats/PortableTestObjects/interface/TestSoA.h"
+#include "HeterogeneousCore/AlpakaInterface/interface/config.h"
+
+namespace ALPAKA_ACCELERATOR_NAMESPACE {
+
+  namespace portabletest {
+
+    // import the top-level portabletest namespace
+    using namespace ::portabletest;
```

Comment:
```
Rule1 : https://raw.githubusercontent.com/cms-sw/cmssw/master/Utilities/ReleaseScripts/python/cmsCodeRules/config.py
Search for "using namespace" or "using std::" in header files

/DataFormats/PortableTestObjects/interface/alpaka/TestDeviceCollection.h
[13]
/AlpakaInterface/interface/config.h
[43, 69, 95, 121]
```
even though in this context it may probably make sense

Reply:
Thanks @perrotta for checking. The main reason for this code rule check is to avoid `using namespace X` to add names into the global namespace. The relevant C++ Core Guideline is https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#sf7-dont-write-using-namespace-at-global-scope-in-a-header-file. This line adds the names into another namespace, which is ok (one could argue our checker is too eager to warn).